src/qbe_ir_generator.cpp

```diff
@@ -205,11 +205,34 @@ void QbeIrGenerator::Visit(const ArrDeclNode& arr_decl) {
   }
 }
 
-void QbeIrGenerator::Visit(const RecordDeclNode& struct_def) {}
+void QbeIrGenerator::Visit(const RecordDeclNode& record_decl) {
+  /* Do nothing because this node only declares a type. */
+}
+
+void QbeIrGenerator::Visit(const FieldNode& field) {
+  /* Do nothing because this node only declares a member type in a record. */
+}
 
-void QbeIrGenerator::Visit(const FieldNode& field) {}
+void QbeIrGenerator::Visit(const RecordVarDeclNode& record_var_decl) {
+  const auto base_addr = NextLocalNum();
+  // TODO: support different data types. We have `int` type for now.
+  WriteInstr_("{} =l alloc4 {}", FuncScopeTemp{base_addr},
+              record_var_decl.type->size());
+  id_to_num[record_var_decl.id] = base_addr;
 
-void QbeIrGenerator::Visit(const RecordVarDeclNode& struct_def) {}
+  for (auto i = std::size_t{0}, e = record_var_decl.inits.size(); i < e; ++i) {
+    const auto& init = record_var_decl.inits.at(i);
+    init->Accept(*this);
+    const auto init_num = num_recorder.NumOfPrevExpr();
+
+    // res_addr = base_addr + offset
+    const int res_addr_num = NextLocalNum();
+    WriteInstr_("{} =l add {}, {}", FuncScopeTemp{res_addr_num},
+                FuncScopeTemp{base_addr}, i * init->type->size());
```

Comment:
Shouldn't we use the `init->type->offset()` here?
We might want to overload the `offset()` function to take the index of the member so that we can support the un-designating version of initialization.

Reply:
> We might want to overload the `offset()` function to take the index of the member so that we can support the un-designating version of initialization.

Can you elaborate on this? 🤔 You mean add a `index()` function to type?

Reply:
`offset(std::size_t index)` if necessary. This depends on how you revise the expression `i * init->type->size()`. I'll elaborate after seeing the change.